### PR TITLE
Fix bug in S40RTS, affects 2 tests

### PR DESCRIPTION
**Pull request by jaustermann**
The option 'zero out degree 0' did not only zero out degree 0 but also every order 0. In the two initial condition S20RTS / S40RTS tests this switch is on so they needed to be updated.

### CODE REVIEW

File: source/initial_conditions/S40RTS_perturbation.cc [hunk 1]
```diff
@@ -265,28 +265,29 @@ namespace aspect
       double prefact;
       int ind = 0;
 
+      // option to not include degree zero in summation
+      unsigned int starting_degree = (zero_out_degree_0
```
Comment (bangerth) [inline source/initial_conditions/S40RTS_perturbation.cc]: make it `const`

File: tests/initial_condition_S20RTS/screen-output [hunk 2]
```diff
@@ -12,28 +12,28 @@ Number of degrees of freedom: 4,030 (2,910+150+970)
 *** Timestep 0:  t=0 seconds
    Solving temperature system... 0 iterations.
    Rebuilding Stokes preconditioner...
-   Solving Stokes system... 30+16 iterations.
+   Solving Stokes system... 30+15 iterations.
 
    Postprocessing:
      Writing graphical output: output-initial_condition_S20RTS/solution-00000
-     Temperature min/avg/max:  1371 K, 1599 K, 1724 K
+     Temperature min/avg/max:  1488 K, 1598 K, 1708 K
```
Comment (bangerth) [inline tests/initial_condition_S20RTS/screen-output]: wow, this *did* make a difference!

**Comment by jaustermann** [comment]
@bangerth updated the const and yes, it did mike a difference! We didn't use that option in the cookbook though, phew.

**Comment by bangerth** [comment]
I think this is good to merge, but the tests fail. @tjhei : Can you take it and update the test outputs?

**Pull request merged**